src/ray/util/ordered_set.h

```diff
@@ -0,0 +1,44 @@
+#ifndef RAY_UTIL_ORDERED_SET_H
+#define RAY_UTIL_ORDERED_SET_H
+
+#include <list>
+#include <unordered_map>
+
+template <typename T>
+class ordered_set {
+ public:
+  explicit ordered_set() {}
```

Comment:
Why `explicit`?

Reply:
I guess it doesn't matter because there are no arguments. I can't remember what the best practices are here.

Reply:
I'd remove it. I'm struggling to find the example but I've come across situations where `explicit` made a difference for default constructors (and only made the class annoying to use), though I'm not sure if they were due to compiler bugs or not.

The best practice is to put `explicit` for constructors that can be called with 1 argument, unless they're intended for conversions.